File: go_algorithms/algorithms/NaivOnArray.go
package algorithms

import (
	"fmt"
	"time"
)

func NaiveOnArray(A [][]int, B [][]int) ([][]int, int64) {

	fmt.Println("Método: naiveOnArray")

	startTime := time.Now().UnixNano()
	N := len(A)
	M := len(B[0])
	P := len(B)

	if N == M && M == P {
		result := make([][]int, N)
		for i := range result {
			result[i] = make([]int, M)
		}

		for i := 0; i < N; i++ {
			for j := 0; j < M; j++ {
				aux := 0
				for k := 0; k < P; k++ {
					aux += A[i][k] * B[k][j]
				}
				result[i][j] = aux
			}
		}

		endTime := time.Now().UnixNano()
		executionTime := endTime - startTime
		return result, executionTime
	} else {
		fmt.Println("Las dimensiones de las matrices no son iguales.")
		return nil, 0
	}
}

File: python_algorithms/algorithms/NaivOnArray.py
import numpy as np
import time

def Multiply(A, B):
    print("Método: naiveOnArray")

    start_time = time.time_ns()
    N = len(A)
    M = len(B[0])
    P = len(B)

    result = np.zeros((N, M), dtype=np.int64)

    for i in range(N):
        for j in range(M):
            aux = 0
            for k in range(P):
                aux += A[i][k] * B[k][j]
            result[i][j] = aux

    end_time = time.time_ns()
    execution_time = end_time - start_time
    return result, execution_time
